tfjs-layers/src/layers/nlp/models/gpt2/gpt2_backbone.ts

```diff
@@ -0,0 +1,252 @@
+import { input } from '../../../../exports';
+import { Embedding } from '../../../embeddings';
+import { SymbolicTensor } from '../../../../engine/topology';
+import { PositionEmbedding } from '../../modeling/position_embedding';
+import { add } from '../../../../exports_layers';
+import { Dropout } from '../../../core';
+import { TransformerDecoder } from '../../modeling/transformer_decoder';
+import { getActivation } from '../../../../activations';
+import { LayerNormalization } from '../../../normalization';
+import { Backbone } from '../backbone';
+
+function gpt2KernelInitializer(stddev = 0.02) {
+  return new RandomNormal({stddev});
+}
+
+export interface GPT2BackboneArgs  {
+  /**
+   * Integer. The size of the token vocabulary.
+   */
+  vocabularySize: number;
+
+  /**
+   * Integer. The number of transformer layers.
+   */
+  numLayers: number;
+
+  /**
+   * Integer. The number of attention heads for each transformer.
+   * The hidden size must be divisible by the number of attention heads.
+   */
+  numHeads: number;
+
+  /**
+   * Integer. The size of the transformer encoding and pooler layers.
+   */
+  hiddenDim: number;
+
+  /**
+   * Integer. The output dimension of the first Dense layer in a two-layer
+   * feedforward network for each transformer.
+   */
+  intermediateDim: number;
+
+  /**
+   * Float. Dropout probability for the Transformer encoder.
+   * Defaults to 0.2.
+   */
+  dropout?: number;
+
+  /**
+   * Integer. The maximum sequence length that this encoder can consume.
+   * If `null`, `maxSequenceLength` uses the value from sequence length.
+   * This determines the variable shape for positional embeddings.
+   * Defaults to 1024.
+   */
+  maxSequenceLength?: number;
+}
+
+/**
+ * GPT-2 core network with hyperparameters.
+ *
+ * This network implements a Transformer-based decoder network,
+ * Generative Pretrained Transformer-2 (GPT-2), as described in
+ * ["Language Models are Unsupervised Multitask Learners"](https://cdn.openai.com/better-language-models/language_models_are_unsupervised_multitask_learners.pdf).
+ * It includes the embedding lookups and transformer layers.
+ *
+ * The default constructor gives a fully customizable, randomly initialized
+ * GPT-2 model with any number of layers, heads, and embedding
+ * dimensions. To load preset architectures and weights, use the `fromPreset`
+ * constructor.
+ *
+ * Disclaimer: Pre-trained models are provided on an "as is" basis, without
+ * warranties or conditions of any kind. The underlying model is provided by a
+ * third party and subject to a separate license, available
+ * [here](https://github.com/openai/gpt-2).
+ *
+ *
+ * Example usage:
+ * ```js
+ * const tokenIds = tf.ones([1, 12]), dtype="int32");
+ * const paddingMask = tf.tensor(
+ *  [[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0]], 'int32');
+ *
+ * # Pretrained GPT-2 decoder.
+ * model = GPT2Backbone.fromPreset("gpt2_base_en");
+ * model.apply(inputData, {paddingMask});
```

Comment:
We might write this part of the API that supports automatic preset loading in tfjs-models. Let's see what makes the most sense as we work on model loading, though.